### ISSUE TITLE
perlbrew: Module 'DynaLoader' is not installed

### ISSUE THREAD

**Issue by wollmers** [perl11/cperl#52]
cperl-5.22.1 installed on OSX 10.9.5 with perlbrew

```
$ cpanm LCS
--> Working on LCS
Fetching http://www.cpan.org/authors/id/W/WO/WOLLMERS/LCS-0.10.tar.gz ... OK
==> Found dependencies: Module::Build::Tiny
--> Working on Module::Build::Tiny
Fetching http://www.cpan.org/authors/id/L/LE/LEONT/Module-Build-Tiny-0.039.tar.gz ... OK
==> Found dependencies: ExtUtils::Helpers, ExtUtils::InstallPaths, DynaLoader, ExtUtils::Config
--> Working on ExtUtils::Helpers
Fetching http://www.cpan.org/authors/id/L/LE/LEONT/ExtUtils-Helpers-0.022.tar.gz ... OK
Configuring ExtUtils-Helpers-0.022 ... OK
Building and testing ExtUtils-Helpers-0.022 ... OK
Successfully installed ExtUtils-Helpers-0.022
--> Working on ExtUtils::InstallPaths
Fetching http://www.cpan.org/authors/id/L/LE/LEONT/ExtUtils-InstallPaths-0.011.tar.gz ... OK
Configuring ExtUtils-InstallPaths-0.011 ... OK
==> Found dependencies: ExtUtils::Config
--> Working on ExtUtils::Config
Fetching http://www.cpan.org/authors/id/L/LE/LEONT/ExtUtils-Config-0.008.tar.gz ... OK
Configuring ExtUtils-Config-0.008 ... OK
Building and testing ExtUtils-Config-0.008 ... OK
Successfully installed ExtUtils-Config-0.008
Building and testing ExtUtils-InstallPaths-0.011 ... OK
Successfully installed ExtUtils-InstallPaths-0.011
skipping N/NW/NWCLARK/perl-5.8.6.tar.gz
! Installing the dependencies failed: Module 'DynaLoader' is not installed
! Bailing out the installation for Module-Build-Tiny-0.039.
! Installing the dependencies failed: Module 'Module::Build::Tiny' is not installed
! Bailing out the installation for LCS-0.10.
3 distributions installed
```

**Comment by rurban** [comment]
Another chicken or egg toolchain problem. 
They don't see the new builtins, like strict, DynaLoader, XSLoader. This seems to be important.

**Comment by wollmers** [comment]
What could be a general solution? A thin wrapper or dummy? Patching popular toolchain modules?

**Comment by rurban** [comment]
In $VERSION either leave out the 'c' or provide proper dualvars (string + number) in the pure-perl modules, or find a better solution.
But strict, DynaLoader and XSLoader $VERSION are already proper dualvars, i.e. they have proper a numeric representation, which is not used by the toolchain. I'll check

**Comment by zdm** [comment]
The same problem under windows, using msys2 mingw-x64 compiler:

```
dmake test

....

..\miniperl.exe -I..\lib ..\autodoc.pl ..
..\miniperl.exe -I..\lib ..\pod\perlmodlib.PL -q ..
..\cperl.exe -f ..\pod\buildtoc -q
Can't locate DynaLoader.pm in @INC (you may need to install the DynaLoader module) (@INC contains: d:/downloads/cperl-cperl-5.22.1/lib .) at d:/downloads/cperl-cperl-5.22.1/lib/Win32.pm line 8.
Compilation failed in require at d:/downloads/cperl-cperl-5.22.1/lib/Cwd.pm line 624.
BEGIN failed--compilation aborted at d:/downloads/cperl-cperl-5.22.1/lib/FindBin.pm line 166.
BEGIN failed--compilation aborted at d:/downloads/cperl-cperl-5.22.1/lib/FindBin.pm line 166.
Compilation failed in require at ..\pod\buildtoc line 6.
BEGIN failed--compilation aborted at ..\pod\buildtoc line 6.
dmake:  Error code 130, while making '..\pod\perltoc.pod'
```

**Comment by wollmers** [comment]
@zdm I would expect it to behave the same on all OSes, because it's a problem of some or many perl modules.

**Comment by zdm** [comment]
@wollmers Is there are any recommendations how I can build it?

**Comment by rurban** [comment]
This should be fixed now with 90eff21b2edbdf93ac9c8c007e520012bb5d3cbf and 0b03dc06288135f81f267e5ce706afdc9851d707
and the various more EUMM, CPAN and CPAN-Meta-Requirements, ... toolchain related enhancements
